rho/host_discovery.py

```diff
@@ -133,12 +135,17 @@ def create_ping_inventory(vault, vault_pass, profile_ranges, profile_port,
     # verbosity can break our parsing of Ansible's output. This is
     # a temporary fix - a better solution would be less-fragile
     # output parsing.
-    ansible_utils.run_with_vault(cmd_string, vault_pass,
-                                 log_path=PING_LOG_PATH,
-                                 env=my_env,
-                                 log_to_stdout=process_discovery_scan,
-                                 log_to_stdout_env=log_env,
-                                 ansible_verbosity=0, error_on_failure=False)
+    with stopit.ThreadingTimeout(30 * 60) as timeout_mgr:
```

Comment:
I anticipated we would use the timeout capabilities in `pexpect.spawn` to accomplish this, instead of introducing another module.
http://pexpect.readthedocs.io/en/stable/api/pexpect.html#spawn-class

Reply:
Oh, I didn't realize pexpect had this built-in. I'll use that.